### DISCUSSION TITLE
How to make some option resilient?

### DISCUSSION THREAD

**Discussion by Asqiir** [nix-community/plasma-manager#92] [Q&A] [answered]
I would like to be able to predefine some options in nix as "whatever value is here should survive a `home-manager switch`".

I imagine it to work like this:

```nix
config.programs.plasma.configFile.<file>.<segement>.<option> = mkResilient;
```

This would be part of the activationScript. It needs to read from the configfile before deletion, and write afterwards.

In some cases, I would prefer this over excluding the entire script from override.

**Comment by toast003** [comment]
I'm not sure what you mean?

Tried looking up what `mkResilient` is but I could not find anything

**Reply by Asqiir** [reply]
The point of this question is "is there already an option that works kinda like this (and of not, what is your opinion on whether there should be such a thing)".

Now that I think of it `persistent` fits better than `resilient` does.

**Reply by toast003** [reply, marked as answer]
Ok, I think what you mean now.
We don't have anything that does this rn, although I could see it being implemented after https://github.com/pjones/plasma-manager/pull/94 lands (no promises tho :P)

**Reply by toast003** [reply]
Imo it should call it mutable instead of persistent (look at `users.mutableUsers` in NixOS), so for example if you wanted to make the amount of desktops mutable you could do something like this:
```nix
programs.plasma.configFile = {
  "kwinrc"."Desktops"."Number" = {
    mutable = true;
  };
};
```

**Reply by magnouvean** [reply]
The reason why I made `immutable` instead of mutable was mainly because I thought it made more sense to make something "immutable" than "not mutable" (mutable sort of is the default). It still can change though if there is some reason to do so. Regarding adding options for making things resilient I guess we could add such an option now as it's more approachable after #94 was merged, though it isn't really that trivial to implement right now (we probably would need to rewrite the config-writing script) and some other things probably is a bit more important a.t.m..
